src/_shared/js/capi-multiple.js

```diff
@@ -0,0 +1,262 @@
+import { getIframeId, getWebfonts, resizeIframeHeight } from
+	'./messages.js';
+import { write } from './dom.js';
+
+const ENDPOINT = 'https://api.nextgen.guardianapps.co.uk/commercial/api/capi-multiple.json';
+
+// Glabs edition links.
+const GLABS_EDITION = {
+	default: 'https://theguardian.com/guardian-labs',
+	au: 'https://theguardian.com/guardian-labs-australia'
+};
+
+const OVERRIDES = {
+	urls: ['[%Article1URL%]', '[%Article2URL%]', '[%Article3URL%]',
+		'[%Article4URL%]'],
+	headlines: ['[%Article1Headline%]', '[%Article2Headline%]',
+		'[%Article3Headline%]', '[%Article4Headline%]'],
+	images: ['[%Article1Image%]', '[%Article2Image%]', '[%Article3Image%]',
+		'[%Article4Image%]'],
+	brandLogo: '[%BrandLogo%]'
+};
+
+// Loads the card data from CAPI in JSON format.
+function retrieveCapiData () {
+
+	let params = new URLSearchParams();
+	params.append('k', '[%SeriesURL%]');
+
+	OVERRIDES.urls.map(url => {
+
+		if (url !== '') {
+			params.append('t', url);
+		}
+
+	})
+
+	let url = `${ENDPOINT}?${params}`;
+
+	return fetch(url).then(response => response.json());
+
+}
+
+// Sets up media icon information on a card (SVG and class).
+function setMediaIcon (card, title, mediaType) {
+
+	title.insertAdjacentHTML('afterbegin', mediaIcons[mediaType]);
+	card.classList.add('advert--media');
+
+}
+
+// Inserts capi headline or DFP override.
+function buildHeadline (title, cardInfo, cardNumber) {
+
+	let headline;
+
+	if (OVERRIDES.headlines[cardNumber] !== '') {
+		headline = document.createTextNode(OVERRIDES.headlines[cardNumber]);
+	} else {
+		headline = document.createTextNode(cardInfo.articleHeadline);
+	}
+
+	title.appendChild(headline);
+
+}
+
+// Constructs the title part of the card: headline and media icon.
+function buildTitle (card, cardInfo, cardNumber) {
+
+	let title = card.querySelector('.advert__title');
+
+	if (cardInfo.videoTag) {
```

Comment:
Relevant commit in frontend [here](https://github.com/guardian/frontend/pull/14473/commits/3f6e7fcf9a118adda6fa99496af9b5c54ea93ca8).